src/routeApp/routes.js

```diff
@@ -0,0 +1,48 @@
+import { Router } from 'express';
+import asyncHandler from '../utils/asyncHandler.js';
+import {
+	getRoutes,
+	getSingleRoute,
+	createRoute,
+	deleteRoute,
+	editRoute,
+} from './controllers.js';
+import routeValidation from './validations.js';
+import validate from '../utils/validateMiddleware.js';
+import verifyToken from '../middlewares/authJwt.js';
+import CheckPermissionAny from '../middlewares/CheckPermission.js';
+
+const routeRouter = Router();
+const resource = 'routes';
+
+routeRouter.get('/', verifyToken, CheckPermissionAny(resource), getRoutes);
+routeRouter.post(
+	'/',
+	verifyToken,
+	CheckPermissionAny(resource),
+	routeValidation(),
+	validate,
+	createRoute
+);
+routeRouter.get(
+	'/:id',
+	verifyToken,
+	CheckPermissionAny(resource),
+	getSingleRoute
+);
+routeRouter.delete(
+	'/:id',
+	verifyToken,
+	CheckPermissionAny(resource),
+	deleteRoute
+);
+routeRouter.put(
+	'/:id',
+	verifyToken,
+	CheckPermissionAny(resource),
+	routeValidation(),
+	validate,
+	asyncHandler(editRoute)
+);
+
```

Comment:
you are using the same validations for creating routes and editing route 
what if someone wants just to edit the second destination alone?